VideoLocker/src/main/java/org/edx/mobile/util/Config.java

```diff
@@ -103,30 +108,30 @@ public boolean isEnabled() {
 
     /**
      * Course Enrollment configuration.
+     *
+     * If TYPE is not "webview", defaults to "native"
      */
     public class EnrollmentConfig {
-        private @SerializedName("ENABLED") boolean mEnabled;
-        private @SerializedName("NATIVE_COURSE_DISCOVERY_ENABLED") boolean mNativeCourseDiscoveryEnabled;
+        private @SerializedName("TYPE") String mCourseEnrollmentType;
         private @SerializedName("COURSE_SEARCH_URL") String mSearchUrl;
-        private @SerializedName("EXTERNAL_COURSE_SEARCH_URL") String mExternalSearchUrl;
         private @SerializedName("COURSE_INFO_URL_TEMPLATE") String mCourseInfoUrlTemplate;
 
-        public boolean isEnabled() {
-            return mEnabled;
-        }
-
-        public boolean isNativeCourseDiscoveryEnabled() {
-            return mNativeCourseDiscoveryEnabled;
+        public boolean isWebviewCourseDiscoveryEnabled() {
+            if (mCourseEnrollmentType != null) {
+                if (mCourseEnrollmentType.equalsIgnoreCase(EnrollmentType.WEBVIEW.name())) {
+                    return true;
+                } else if (!mCourseEnrollmentType.equalsIgnoreCase(EnrollmentType.NATIVE.name())) {
+                    // logger.debug("No match in config for COURSE_ENROLLMENT.TYPE:" + mCourseEnrollmentType + ". Defaulting to Native");
```

Comment:
All of these inner classes should be defined as static, as Gson falls back to attempting to initialize them in an unsafe manner if there is no instance creator registered (just like any other class that doesn't define the default no-args constructor), even though the docs [state](https://sites.google.com/site/gson/gson-user-guide#TOC-Nested-Classes-including-Inner-Classes-) that this is not supported. Thus they are effectively initialized as static classes anyway, and any attempt to reference the outer class will result in a `NullPointerException`.